src/prompt-input/interfaces.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { BaseComponentProps } from '../internal/base-component';
+import { BaseKeyDetail, NonCancelableEventHandler } from '../internal/events';
+import {
+  BaseInputProps,
+  InputAutoCorrect,
+  InputAutoComplete,
+  InputKeyEvents,
+  InputSpellcheck,
+  BaseChangeDetail,
+} from '../input/interfaces';
+import { FormFieldValidationControlProps } from '../internal/context/form-field-context';
+import { IconProps } from '../icon/interfaces';
+
+export interface PromptInputProps
```

Comment:
another general thought: for the sake of future archaeologists to follow up in `quip/vQJVAwOM2gv2` with some comments/answers regarding  decisions that were made here. For example, about `select()`

Reply:
This is a great catch! I missed that method in the API doc, since Ruben hadn't implemented it already. I've added it.

Reply:
I've also updated the API doc with a couple of the other changes we made, namely changing `submit` to `action` everywhere you see it.